internal/pkg/agent/application/monitoring/v1_monitor.go

```diff
@@ -616,249 +616,137 @@ func (b *BeatsMonitor) injectMetricsInput(
 		failureThreshold = &defaultValue
 	}
 	monitoringNamespace := b.monitoringNamespace()
-	fixedAgentName := strings.ReplaceAll(agentName, "-", "_")
+
+	//create a new map with the monitoring beats included
+	componentListWithMonitoring := map[string]string{
+		fmt.Sprintf("beat/%s", monitoringMetricsUnitID): "metricbeat",
+		fmt.Sprintf("http/%s", monitoringMetricsUnitID): "metricbeat",
+		monitoringFilesUnitsID:                          "filebeat",
+	}
+	for k, v := range componentIDToBinary {
+		componentListWithMonitoring[k] = v
+	}
+
 	// beatStreams and streams MUST be []interface{} even if in reality they are []map[string]interface{}:
 	// if those are declared as slices of maps the message "proto: invalid type: []map[string]interface{}" will pop up
-	beatsStreams := make([]interface{}, 0, len(componentIDToBinary))
-	streams := []interface{}{
+	beatsStreams := b.getBeatsStreams(componentListWithMonitoring, failureThreshold, metricsCollectionIntervalString)
+	httpStreams := b.getHttpStreams(componentListWithMonitoring, failureThreshold, metricsCollectionIntervalString)
+
+	inputs := []interface{}{
 		map[string]interface{}{
-			idKey: fmt.Sprintf("%s-agent", monitoringMetricsUnitID),
+			idKey:        fmt.Sprintf("%s-beats", monitoringMetricsUnitID),
+			"name":       fmt.Sprintf("%s-beats", monitoringMetricsUnitID),
+			"type":       "beat/metrics",
```

Comment:
Was this change from `%s-agent` to `%s-beats` intentional?  Note that, further below, when setting up the "http/metrics" type input, it's still `%s-agent` there.